apps/molgenis-components/src/components/tables/RefTable.vue

```diff
@@ -1,66 +1,119 @@
       </span>
+    </h5>
+    <div>
+      <div :class="{ 'collapsed-table': collapsed }">
+        <table class="table table-sm mb-2">
+          <tr v-for="(value, key) in filteredResults(reference)">
+            <td class="key border-right">{{ key }}</td>
+            <td class="value">
+              <div v-if="reference.metadata">
+                <DataDisplayCell
+                  :data="value"
+                  :meta-data="metadataOfRow(key)"
+                />
+              </div>
+              <div v-else>{{ value }}</div>
+            </td>
+          </tr>
+        </table>
+      </div>
+      <div
+        v-if="collapsed"
+        class="collapsed-tag border-top rounded-bottom mb-3"
+        @click="collapsed = false"
+      >
+        <small class="px-3 link-color"> Show all records... </small>
+      </div>
     </div>
-    <div v-if="reference.mg_updatedBy">
-      Updated by '{{ reference.mg_updatedBy }}'
-      <span v-if="reference.mg_updatedOn">
-        On {{ new Date(reference.mg_updatedOn as string).toLocaleString() }}
-      </span>
+
+    <small class="text-black-50" v-if="showDataOwner">
+      <div v-if="reference.mg_insertedBy">
+        Inserted by '{{ reference.mg_insertedBy }}'
+        <span v-if="reference.mg_insertedOn">
+          On {{ new Date(reference.mg_insertedOn as string).toLocaleString() }}
+        </span>
+      </div>
+      <div v-if="reference.mg_updatedBy">
+        Updated by '{{ reference.mg_updatedBy }}'
+        <span v-if="reference.mg_updatedOn">
+          On {{ new Date(reference.mg_updatedOn as string).toLocaleString() }}
+        </span>
+      </div>
+    </small>
+    <div v-if="reference.mg_draft">
+      <span class="badge badge-secondary">Draft</span>
     </div>
-  </small>
-  <div v-if="reference.mg_draft">
-    <span class="badge badge-secondary">Draft</span>
   </div>
 </template>
 
 <style scoped>
 table .key {
   width: 0;
 }
+.collapsed-table {
+  max-height: 6.5rem;
+  overflow: hidden;
+}
+.collapsed-tag {
+  text-align: center;
+  color: var(--primary);
+  cursor: pointer;
+}
+.collapsed-tag:hover {
+  text-decoration: underline;
+}
+.collapsed-table .table {
+  margin-bottom: 0 !important;
+}
 </style>
 
 <script lang="ts" setup>
+import { ref } from "vue";
 import { IRefModalData } from "../../Interfaces/IRefModalData";
 import { ITableMetaData } from "../../Interfaces/ITableMetaData";
+import { getPrimaryKey } from "../utils";
 import DataDisplayCell from "./DataDisplayCell.vue";
 
-const props = defineProps<{
+const { reference, startsCollapsed } = defineProps<{
   reference: IRefModalData;
   showDataOwner?: boolean;
+  startsCollapsed?: boolean;
 }>();
 
-function filteredResults(reference: IRefModalData): IRefModalData {
-  const filtered = { ...reference };
-  delete filtered.name;
+const canCollapse = Object.keys(filteredResults(reference)).length > 3;
+let collapsed = ref(startsCollapsed);
+if (collapsed && !canCollapse) {
+  collapsed.value = false;
+}
```

Comment:
this should be part of the initialisation e.g. ref(startCollapsed && canCollapse);